e2e-tests/smart-update/run

```diff
@@ -0,0 +1,197 @@
+    sleep 7 # wait for two reconcile loops ;)  3 sec x 2 times + 1 sec = 7 seconds
+
+    check_last_pod_to_update ${cluster} ${initial_primary} ${pxc_size} ${IMAGE_PXC_TO_UPDATE}
+    wait_cluster_consistency ${cluster} ${pxc_size} $(kubectl_bin get pxc ${cluster} -o jsonpath='{.spec.'$(get_proxy_engine ${cluster})'.size}')
+    for i in $(seq 0 $((${pxc_size} - 1))); do
+        compare_mysql_cmd "select-1" "SELECT * from myApp.myApp;" "-h $cluster-pxc-$i.$cluster-pxc -uroot -proot_password"
+    done
+
+    kubectl_bin delete -f "$test_dir/conf/$cluster-haproxy.yml"
+    kubectl_bin delete pvc --all
+
+    desc 'Updating PXC cluster with version service available but disabled'
+    spinup_pxc "$cluster" "$test_dir/conf/$cluster-version-service-reachable.yml"
+    pxc_size=$(kubectl_bin get pxc "${cluster}" -o jsonpath='{.spec.pxc.size}')
+
+    initial_primary=$(run_mysql 'SELECT @@hostname hostname;' "-h ${cluster}-haproxy -uroot -proot_password")
+    kubectl_bin patch pxc/$cluster --type=merge -p '{"spec":{"pxc":{"image":"'"${IMAGE_PXC_TO_UPDATE}"'"}}}'
+    sleep 7 # wait for two reconcile loops ;)  3 sec x 2 times + 1 sec = 7 seconds
+
+    check_last_pod_to_update ${cluster} ${initial_primary} ${pxc_size} ${IMAGE_PXC_TO_UPDATE}
+
+    wait_cluster_consistency ${cluster} ${pxc_size} $(kubectl_bin get pxc ${cluster} -o jsonpath='{.spec.'$(get_proxy_engine ${cluster})'.size}')
+    for i in $(seq 0 $((${pxc_size} - 1))); do
+        compare_mysql_cmd "select-1" "SELECT * from myApp.myApp;" "-h ${cluster}-pxc-${i}.${cluster}-pxc -uroot -proot_password"
+    done
+
+    kubectl_bin delete -f $test_dir/conf/$cluster-version-service-reachable.yml
+    kubectl_bin delete pvc --all
+
+    desc 'PXC cluster with version service offline'
+    spinup_pxc "$cluster" "$test_dir/conf/$cluster-version-service-unreachable.yml"
+    pxc_size=$(kubectl_bin get pxc "${cluster}" -o jsonpath='{.spec.pxc.size}')
+
+    wait_cluster_consistency ${cluster} ${pxc_size} $(kubectl_bin get pxc ${cluster} -o jsonpath='{.spec.'$(get_proxy_engine ${cluster})'.size}')
+    if [[ $(kubectl_bin get pxc/${cluster} -o jsonpath='{.spec.pxc.image}') != ${IMAGE_PXC} ]]; then
+        echo "ERROR: PXC image has been changed. Exiting..."
+        exit 1
+    fi
+
+    desc 'PXC cluster update with recommended image by version service'
+    pxc_size=$(kubectl_bin get pxc "${cluster}" -o jsonpath='{.spec.pxc.size}')
+    version_service_image="recommended"
+    initial_primary=$(run_mysql 'SELECT @@hostname hostname;' "-h $cluster-haproxy -uroot -proot_password")
+
+    kubectl_bin patch pxc/$cluster --type=merge -p '{"spec":{"upgradeOptions":{"versionServiceEndpoint":"https://check.percona.com/versions","apply":"'${version_service_image}'","schedule": "* * * * *"}}}'
+    target_image=$(get_vs_image ${version_service_image})
+    sleep 55
+
+    check_last_pod_to_update ${cluster} ${initial_primary} ${pxc_size} ${target_image}
```

Comment:
version service provides all images (haproxy, proxysql, backup, pxc), so not just PXC, maybe it would be good to check if all of them were actually upgraded on the cluster